src/topology/class.c

```diff
@@ -21,13 +21,43 @@
 #include "tplg2_local.h"
 #include <ctype.h>
 
+static int tplg_parse_class_attribute(snd_tplg_t *tplg ATTRIBUTE_UNUSED, snd_config_t *cfg,
+				      struct tplg_attribute *attr)
+{
+	snd_config_iterator_t i, next;
+	snd_config_t *n;
+	const char *id;
+
+	snd_config_for_each(i, next, cfg) {
+		n = snd_config_iterator_entry(i);
+
+		if (snd_config_get_id(n, &id) < 0)
+			continue;
+
+		/* Parse the tuple token reference */
+		if (!strcmp(id, "token_ref")) {
+			const char *s;
+
+			if (snd_config_get_string(n, &s) < 0) {
+				SNDERR("invalid token_ref for attribute %s\n", attr->name);
+				return -EINVAL;
```

Comment:
can we get file and line no here ?

Reply:
Actually, yes, the line number and file are already printed with SNDERR.